resources/metadata/gitopsUpdateDeployment.yaml

```diff
@@ -0,0 +1,112 @@
+metadata:
+  name: gitopsUpdateDeployment
+  description: Updates Kubernetes Deployment Manifest in Infrastructure Git Repo
+  longDescription: |
+    This step allows you to update the deployment manifest in another repository.
+
+    It can for example be used for GitOps scenarios where the update of the manifests triggers an update of the corresponding deployment in Kubernetes.
+spec:
+  inputs:
+    secrets:
+      - name: gitHttpsCredentialsId
+        description: Jenkins 'Username with password' credentials ID containing username/password for http access to your git repository.
+        type: jenkins
+    params:
+      - name: branchName
+        description: Currently ignored and only the main branch is considered. The name of the branch you want the changes pulled into.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        mandatory: true
+      - name: commitMessage
+        description: The commit message of the commit that will be done to do the changes.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        mandatory: true
+      - name: serverUrl
+        aliases:
+          - name: githubServerUrl
+        description: GitHub server url to the repository.
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        default: https://github.com
+        mandatory: true
+      - name: username
+        type: string
+        description: User name for git authentication
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        secret: true
+        resourceRef:
+          - name: gitHttpsCredentialsId
+            type: secret
+            param: username
+      - name: password
+        type: string
+        description: Password/token for git authentication.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        secret: true
+        resourceRef:
+          - name: gitHttpsCredentialsId
+            type: secret
+            param: password
+      - name: filePath
```

Comment:
What about specifying this a bit more, e.g. name it `deploymentDescriptor`?

Reply:
should the name contain the word "path" as it requires the whole path in the git repo to the manifest

Reply:
Yes, I think we should have `path` in there. Not sure if we should make it too concrete for now.
I would be okay with `filePath`
We need to see what we need when adding `helm`, `kustomize`, ... how it looks like then ...